cpp/src/arrow/compare.cc

```diff
@@ -1030,8 +1030,10 @@ Status PrintDiff(const Array& left, const Array& right, int64_t left_offset,
   }
 
   if (!left.type()->Equals(right.type())) {
-    *os << "# Array types differed: " << *left.type() << " vs " << *right.type()
-        << std::endl;
+    *os << "# Array types differed:" << std::endl
+        << *left.type() << std::endl
+        << "vs" << std::endl
+        << *right.type() << std::endl;
```

Comment:
Before (left type and right type aren't aligned):

```text
# Array types differed: ${left type} vs ${right type}
```

After (left type and right type are aligned):

```text
# Array types differed:
${left type}
vs
${right type}
```

Aligned output is easy to compare.

Reply:
Hmm. It breaks an existing test. I'll revert this in this PR.